src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -786,12 +787,24 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
         manager.register_pass<ov::intel_gpu::ConvertMatMulToFullyConnected>();
         manager.register_pass<ov::intel_gpu::MoveFCReshapeToWeights>();
         manager.register_pass<ov::intel_gpu::ConvertFullyConnectedToFullyConnectedCompressed>(device_info.supports_immad);
+
+        bool disable_horizontal_fc_fusion = false;
+        GPU_DEBUG_GET_INSTANCE(debug_config);
+        GPU_DEBUG_IF(debug_config->disable_horizontal_fc_fusion == 1)
+            disable_horizontal_fc_fusion = true;
+
+        if (!disable_horizontal_fc_fusion)
+            manager.register_pass<ov::intel_gpu::FullyConnectedHorizontalFusion>();
         if (device_info.supports_immad) {
             // For OneDNN, ZP should not be folded for FC. But still, ZP should be folded for Gather.
             // Therefore, run MarkDequantizationSubgraph again to fold ZP constant.
             manager.register_pass<ov::pass::MarkDequantizationSubgraph>(supported_woq_types, true);
-            manager.register_pass<ov::pass::ConstantFolding>();
+            if (disable_horizontal_fc_fusion)
+                manager.register_pass<ov::pass::ConstantFolding>();
         }
+        if (!disable_horizontal_fc_fusion)
+            manager.register_pass<ov::pass::ConstantFolding>();
```

Comment:
What is the impact on model loading time?

Reply:
I checked only for llama3, and there was not obvious loading time increase.
As I marked, will check other network's performance tomorrow on various platforms.